sysboundary/ionosphere.cpp

```diff
@@ -1103,7 +1105,159 @@ namespace SBC {
+      _r[2]=r[2]+( (3./40.)*kz[0] +(9./40.)*kz[1] ); 
+      BFieldFunction(_r,outwards,b_unit);
+      kx[2]=step*b_unit[0];
+      ky[2]=step*b_unit[1];
+      kz[2]=step*b_unit[2];
+      
+      //K4 slope
+      _r[0]=r[0]+(44./45.)*kx[0] -(56./15.)*kx[1] + (32./9.)*kx[2];  
+      _r[1]=r[1]+(44./45.)*ky[0] -(56./15.)*ky[1] + (32./9.)*ky[2];  
+      _r[2]=r[2]+(44./45.)*kz[0] -(56./15.)*kz[1] + (32./9.)*kz[2];  
+      BFieldFunction(_r,outwards,b_unit);
+      kx[3]=step*b_unit[0];
+      ky[3]=step*b_unit[1];
+      kz[3]=step*b_unit[2];
+            
+      //K5 slope  
+      _r[0]=r[0]+(19372./6561.)*kx[0] -(25360./2187.)*kx[1] + (64448./6561.)*kx[2] - (212./729.)*kx[3];   
+      _r[1]=r[1]+(19372./6561.)*ky[0] -(25360./2187.)*ky[1] + (64448./6561.)*ky[2] - (212./729.)*ky[3];   
+      _r[2]=r[2]+(19372./6561.)*kz[0] -(25360./2187.)*kz[1] + (64448./6561.)*kz[2] - (212./729.)*kz[3];   
+      BFieldFunction(_r,outwards,b_unit); 
+      kx[4]=step*b_unit[0];
+      ky[4]=step*b_unit[1];
+      kz[4]=step*b_unit[2];
+            
+      //K6 slope
+      _r[0]=r[0]+(9017./3168.)*kx[0] -(355./33.)*kx[1] - (46732./5247.)*kx[2] + (49./176.)*kx[3]-(5103./18656.)*kx[4];  
+      _r[1]=r[1]+(9017./3168.)*ky[0] -(355./33.)*ky[1] - (46732./5247.)*ky[2] + (49./176.)*ky[3]-(5103./18656.)*ky[4];  
+      _r[2]=r[2]+(9017./3168.)*kz[0] -(355./33.)*kz[1] - (46732./5247.)*kz[2] + (49./176.)*kz[3]-(5103./18656.)*kz[4];  
+      BFieldFunction(_r,outwards,b_unit);
+      kx[5]=step*b_unit[0];
+      ky[5]=step*b_unit[1];
+      kz[5]=step*b_unit[2];
+      
+      //K7 slope 
+      _r[0]=r[0]+(35./384.)*kx[0] + (500./1113.)*kx[2] + (125./192.)*kx[3]-(2187./6784.)*kx[4] +(11./84.)*kx[5];  
+      _r[1]=r[1]+(35./384.)*ky[0] + (500./1113.)*ky[2] + (125./192.)*ky[3]-(2187./6784.)*ky[4] +(11./84.)*ky[5];  
+      _r[2]=r[2]+(35./384.)*kz[0] + (500./1113.)*kz[2] + (125./192.)*kz[3]-(2187./6784.)*kz[4] +(11./84.)*kz[5];  
+      BFieldFunction(_r,outwards,b_unit);
+      kx[6]=step*b_unit[0];
+      ky[6]=step*b_unit[1];
+      kz[6]=step*b_unit[2];
+   
+   
+      //Error calculation
+      std::array<Real,3>rf;
+      std::array<Real,3>error_xyz;
+      rf[0]=r[0] +(35./384.)*kx[0] + (500./1113.)*kx[2] + (125./192.)*kx[3] - (2187./6784.)*kx[4] +(11./84.)*kx[5];
+      rf[1]=r[1] +(35./384.)*ky[0] + (500./1113.)*ky[2] + (125./192.)*ky[3] - (2187./6784.)*ky[4] +(11./84.)*ky[5];
+      rf[2]=r[2] +(35./384.)*kz[0] + (500./1113.)*kz[2] + (125./192.)*kz[3] - (2187./6784.)*kz[4] +(11./84.)*kz[5];
+      
+      error_xyz[0]=abs((71./57600.)*kx[0] -(71./16695.)*kx[2] + (71./1920.)*kx[3] -(17253./339200.)*kx[4]+(22./525.)*kx[5] -(1./40.)*kx[6] );
+      error_xyz[1]=abs((71./57600.)*ky[0] -(71./16695.)*ky[2] + (71./1920.)*ky[3] -(17253./339200.)*ky[4]+(22./525.)*ky[5] -(1./40.)*ky[6] );
+      error_xyz[2]=abs((71./57600.)*kz[0] -(71./16695.)*kz[2] + (71./1920.)*kz[3] -(17253./339200.)*kz[4]+(22./525.)*kz[5] -(1./40.)*kz[6] );
```

Comment:
These *still* look like proper vectorization (using agner's vectorclass) would benefit them. Or are you sure that the compiler actually manags to autovectorize those?

Reply:
Have not checked but I can use vectorclass here indeed.